File: pgSdkSampleJavaApp/src/main/java/com/example/pgsdksamplejavaapp/ProfilesAdapter.java
package com.example.pgsdksamplejavaapp;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

/**
 * Adapter that handles visual representation of the list of configuration profiles.
 */
class ProfilesAdapter extends RecyclerView.Adapter<ProfileViewHolder> {
    private ArrayList<ProfileUiData> profilesList;
    private ProfileClickListener onProfileClicked;

    ProfilesAdapter(ArrayList<ProfileUiData> profiles, ProfileClickListener profileClickListener) {
        profilesList = profiles;
        onProfileClicked = profileClickListener;
    }

    @Override
    @NonNull
    public ProfileViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View profileView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_profile, parent, false);
        return new ProfileViewHolder(profileView);
    }

    @Override
    public void onBindViewHolder(ProfileViewHolder holder, int position) {
        holder.bind(
                profilesList.get(position),
                onProfileClicked
        );
    }

    @Override
    public int getItemCount() {
        return profilesList.size();
    }

    /**
     * Updates list of configuration profiles and refreshes all the items.
     */
    void updateProfiles(ArrayList<ProfileUiData> profiles) {
        profilesList.clear();
        profilesList.addAll(profiles);
        notifyDataSetChanged();
    }
}

/**
 * ViewHolder for displaying configuration profile's overview.
 */
class ProfileViewHolder extends RecyclerView.ViewHolder {

    private Button profileActivateButton;
    private TextView profileActiveLabel;

    ProfileViewHolder(View view) {
        super(view);
        profileActivateButton = view.findViewById(R.id.profileActivateButton);
        profileActiveLabel = view.findViewById(R.id.profileActiveLabel);
    }

    void bind(final ProfileUiData profile, final ProfileClickListener profileClickListener) {
        profileActivateButton.setText(profile.profileId);
        profileActivateButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                profileClickListener.onProfileClicked(profile.profileId);
            }
        });

        if (profile.active) {
            profileActiveLabel.setVisibility(View.VISIBLE);
        } else {
            profileActiveLabel.setVisibility(View.INVISIBLE);
        }
    }
}

File: pgSdkSampleApp/src/main/java/de/proglove/example/sdk/ProfilesAdapter.kt
package de.proglove.example.sdk

import android.view.LayoutInflater
import android.view.View
import android.view.View.INVISIBLE
import android.view.View.VISIBLE
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

/**
 * Adapter that handles visual representation of the list of configuration profiles.
 */
class ProfilesAdapter(
        private val profilesList: MutableList<ProfileUiData> = mutableListOf(),
        private val onProfileClicked: (String) -> Unit
) : RecyclerView.Adapter<ProfileViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProfileViewHolder {
        val profileView =
                LayoutInflater.from(parent.context).inflate(R.layout.item_profile, parent, false)
        return ProfileViewHolder(profileView)
    }

    override fun onBindViewHolder(holder: ProfileViewHolder, position: Int) {
        holder.bind(
                profile = profilesList[position],
                onProfileClicked = onProfileClicked
        )
    }

    override fun getItemCount(): Int {
        return profilesList.size
    }

    /**
     * Updates list of configuration profiles and refreshes all the items.
     */
    fun updateProfiles(profiles: List<ProfileUiData>) {
        profilesList.clear()
        profilesList.addAll(profiles)
        notifyDataSetChanged()
    }
}

/**
 * ViewHolder for displaying configuration profile's overview.
 */
class ProfileViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    private val profileActivateButton: Button = view.findViewById(R.id.profileActivateButton)
    private val profileActiveLabel: TextView = view.findViewById(R.id.profileActiveLabel)

    fun bind(profile: ProfileUiData, onProfileClicked: (String) -> Unit) {
        profileActivateButton.apply {
            text = profile.profileId
            setOnClickListener {
                onProfileClicked(profile.profileId)
            }
        }

        if (profile.active) {
            profileActiveLabel.visibility = VISIBLE
        } else {
            profileActiveLabel.visibility = INVISIBLE
        }
    }
}
